File: backend_foodsave/src/main/java/com/foodsave/backend/domain/enums/Permission.java
package com.foodsave.backend.domain.enums;

public enum Permission {
    // User permissions
    USER_CREATE("user:create"),
    USER_READ("user:read"),
    USER_UPDATE("user:update"),
    USER_DELETE("user:delete"),
    USER_MANAGE("user:manage"),
    
    // Store permissions
    STORE_CREATE("store:create"),
    STORE_READ("store:read"),
    STORE_UPDATE("store:update"),
    STORE_DELETE("store:delete"),
    STORE_MANAGE("store:manage"),
    
    // Product permissions
    PRODUCT_CREATE("product:create"),
    PRODUCT_READ("product:read"),
    PRODUCT_UPDATE("product:update"),
    PRODUCT_DELETE("product:delete"),
    PRODUCT_MANAGE("product:manage"),
    
    // Order permissions
    ORDER_CREATE("order:create"),
    ORDER_READ("order:read"),
    ORDER_UPDATE("order:update"),
    ORDER_DELETE("order:delete"),
    ORDER_MANAGE("order:manage"),
    
    // Analytics permissions
    ANALYTICS_READ("analytics:read"),
    ANALYTICS_EXPORT("analytics:export"),
    ANALYTICS_MANAGE("analytics:manage"),
    
    // Review permissions
    REVIEW_CREATE("review:create"),
    REVIEW_READ("review:read"),
    REVIEW_UPDATE("review:update"),
    REVIEW_DELETE("review:delete"),
    REVIEW_MANAGE("review:manage"),
    
    // Discount permissions
    DISCOUNT_CREATE("discount:create"),
    DISCOUNT_READ("discount:read"),
    DISCOUNT_UPDATE("discount:update"),
    DISCOUNT_DELETE("discount:delete"),
    DISCOUNT_MANAGE("discount:manage"),
    
    // Category permissions
    CATEGORY_CREATE("category:create"),
    CATEGORY_READ("category:read"),
    CATEGORY_UPDATE("category:update"),
    CATEGORY_DELETE("category:delete"),
    CATEGORY_MANAGE("category:manage"),

    // Role permissions
    ROLE_CREATE("role:create"),
    ROLE_READ("role:read"),
    ROLE_UPDATE("role:update"),
    ROLE_DELETE("role:delete"),
    ROLE_MANAGE("role:manage"),

    // Settings permissions
    SETTINGS_READ("settings:read"),
    SETTINGS_UPDATE("settings:update"),
    SETTINGS_MANAGE("settings:manage"),

    // Notification permissions
    NOTIFICATION_CREATE("notification:create"),
    NOTIFICATION_READ("notification:read"),
    NOTIFICATION_UPDATE("notification:update"),
    NOTIFICATION_DELETE("notification:delete"),
    NOTIFICATION_MANAGE("notification:manage");

    private final String value;

    Permission(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
} 

File: admin_website/src/types/permission.ts
export enum Permission {
    // User Management
    USER_READ = 'USER_READ',
    USER_CREATE = 'USER_CREATE',
    USER_UPDATE = 'USER_UPDATE',
    USER_DELETE = 'USER_DELETE',

    // Role Management
    ROLE_READ = 'ROLE_READ',
    ROLE_CREATE = 'ROLE_CREATE',
    ROLE_UPDATE = 'ROLE_UPDATE',
    ROLE_DELETE = 'ROLE_DELETE',

    // Store Management
    STORE_READ = 'STORE_READ',
    STORE_CREATE = 'STORE_CREATE',
    STORE_UPDATE = 'STORE_UPDATE',
    STORE_DELETE = 'STORE_DELETE',

    // Product Management
    PRODUCT_READ = 'PRODUCT_READ',
    PRODUCT_CREATE = 'PRODUCT_CREATE',
    PRODUCT_UPDATE = 'PRODUCT_UPDATE',
    PRODUCT_DELETE = 'PRODUCT_DELETE',

    // Order Management
    ORDER_READ = 'ORDER_READ',
    ORDER_CREATE = 'ORDER_CREATE',
    ORDER_UPDATE = 'ORDER_UPDATE',
    ORDER_DELETE = 'ORDER_DELETE',

    // Category Management
    CATEGORY_READ = 'CATEGORY_READ',
    CATEGORY_CREATE = 'CATEGORY_CREATE',
    CATEGORY_UPDATE = 'CATEGORY_UPDATE',
    CATEGORY_DELETE = 'CATEGORY_DELETE',

    // Cart Management
    CART_READ = 'CART_READ',
    CART_UPDATE = 'CART_UPDATE',

    // Analytics
    ANALYTICS_READ = 'ANALYTICS_READ',
    ANALYTICS_EXPORT = 'ANALYTICS_EXPORT',

    // Discount Management
    DISCOUNT_READ = 'DISCOUNT_READ',
    DISCOUNT_CREATE = 'DISCOUNT_CREATE',
    DISCOUNT_UPDATE = 'DISCOUNT_UPDATE',
    DISCOUNT_DELETE = 'DISCOUNT_DELETE',

    // Review Management
    REVIEW_READ = 'REVIEW_READ',
    REVIEW_CREATE = 'REVIEW_CREATE',
    REVIEW_UPDATE = 'REVIEW_UPDATE',
    REVIEW_DELETE = 'REVIEW_DELETE',

    // System Settings
    SETTINGS_READ = 'SETTINGS_READ',
    SETTINGS_UPDATE = 'SETTINGS_UPDATE',
}

export interface PermissionDescription {
    permission: Permission;
    description: string;
}

export const PERMISSION_DESCRIPTIONS: Record<Permission, string> = {
    // User Management
    [Permission.USER_READ]: 'Просмотр списка пользователей и их профилей',
    [Permission.USER_CREATE]: 'Создание новых пользователей',
    [Permission.USER_UPDATE]: 'Редактирование данных пользователей',
    [Permission.USER_DELETE]: 'Удаление пользователей',

    // Role Management
    [Permission.ROLE_READ]: 'Просмотр списка ролей и их прав',
    [Permission.ROLE_CREATE]: 'Создание новых ролей',
    [Permission.ROLE_UPDATE]: 'Редактирование прав ролей',
    [Permission.ROLE_DELETE]: 'Удаление ролей',

    // Store Management
    [Permission.STORE_READ]: 'Просмотр списка магазинов и их деталей',
    [Permission.STORE_CREATE]: 'Создание новых магазинов',
    [Permission.STORE_UPDATE]: 'Редактирование данных магазинов',
    [Permission.STORE_DELETE]: 'Удаление магазинов',

    // Product Management
    [Permission.PRODUCT_READ]: 'Просмотр списка продуктов и их деталей',
    [Permission.PRODUCT_CREATE]: 'Создание новых продуктов',
    [Permission.PRODUCT_UPDATE]: 'Редактирование данных продуктов',
    [Permission.PRODUCT_DELETE]: 'Удаление продуктов',

    // Order Management
    [Permission.ORDER_READ]: 'Просмотр списка заказов и их деталей',
    [Permission.ORDER_CREATE]: 'Создание новых заказов',
    [Permission.ORDER_UPDATE]: 'Редактирование статуса заказов',
    [Permission.ORDER_DELETE]: 'Удаление заказов',

    // Category Management
    [Permission.CATEGORY_READ]: 'Просмотр списка категорий',
    [Permission.CATEGORY_CREATE]: 'Создание новых категорий',
    [Permission.CATEGORY_UPDATE]: 'Редактирование категорий',
    [Permission.CATEGORY_DELETE]: 'Удаление категорий',

    // Cart Management
    [Permission.CART_READ]: 'Просмотр корзин пользователей',
    [Permission.CART_UPDATE]: 'Управление корзинами',

    // Analytics
    [Permission.ANALYTICS_READ]: 'Просмотр аналитики и отчетов',
    [Permission.ANALYTICS_EXPORT]: 'Экспорт данных аналитики',

    // Discount Management
    [Permission.DISCOUNT_READ]: 'Просмотр списка скидок',
    [Permission.DISCOUNT_CREATE]: 'Создание новых скидок',
    [Permission.DISCOUNT_UPDATE]: 'Редактирование скидок',
    [Permission.DISCOUNT_DELETE]: 'Удаление скидок',

    // Review Management
    [Permission.REVIEW_READ]: 'Просмотр отзывов',
    [Permission.REVIEW_CREATE]: 'Создание отзывов',
    [Permission.REVIEW_UPDATE]: 'Редактирование отзывов',
    [Permission.REVIEW_DELETE]: 'Удаление отзывов',

    // System Settings
    [Permission.SETTINGS_READ]: 'Просмотр системных настроек',
    [Permission.SETTINGS_UPDATE]: 'Изменение системных настроек',
};